torchbenchmark/util/model.py

```diff
@@ -286,6 +286,30 @@ def invoke(self) -> Optional[Tuple[torch.Tensor]]:
                 out = self.eval()
         return out
 
+    def get_module(self):
+        raise NotImplementedError("To enroll into more granular benchmarking, every BenchmarkModel is exhorted "
+                                  "to implement a way to retrieve a torch module from it. It is okay if this "
+                                  "remains unimplemented for models that have vastly different architectures.")
+
+    # Should return loss
```

Comment:
Maybe we should clarify in the comment or error message that to implement the train loop, user can choose either implementing 1) `train()`, or 2) `forward()` `backward()` `optimizer_step()`.

It might also be good to add `get_optimizer()` and `set_optimizer()` stub here?

Reply:
Good idea, I'll add that :D